File: android/src/com/dontflymoney/auth/File.java
package com.dontflymoney.auth;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

import android.content.Context;

public class File
{
	private static final String extension = ".d59";
	public static final String Ticket = "ticket" + extension;
	
	private Context context;
	private String name;

	private String status;
	
	
	
	public File(Context context, String name)
	{
		this.context = context;
		this.name = name;
	}
	
	
	public void WriteToFile(String data)
	{
	    try
	    {
	    	FileOutputStream fileOutput = context.openFileOutput(name, Context.MODE_PRIVATE);
	    	
	        OutputStreamWriter writer = new OutputStreamWriter(fileOutput);
	        
	        writer.write(data);
	        writer.close();
	    }
	    catch (IOException e)
	    {
	        status = "File write failed: " + e.toString();
	    } 
	}


	public String ReadFromFile()
	{
	    String result = "";

	    try
	    {
	        InputStream inputStream = context.openFileInput(name);

	        if ( inputStream != null )
	        {
	            InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
	            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
	            
	            String line = "";
	            StringBuilder allContent = new StringBuilder();

	            while ( (line = bufferedReader.readLine()) != null )
	            {
	            	allContent.append(line);
	            }

	            inputStream.close();
	            
	            result = allContent.toString();
	        }
	    }
	    catch (FileNotFoundException e)
	    {
	        status = "File not found: " + e.toString();
	    } 
	    catch (IOException e) 
	    {
	        status = "Can not read file: " + e.toString();
	    }

	    return result;
	}
	
	
	
	public String GetStatus()
	{
		return status;
	}
	
}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/io/File.kt
package com.darakeon.dfm.io

import android.content.Context
import com.darakeon.dfm.R
import java.io.*

/*
TODO: use kotlin IO
class File(private val context: Context, name: FileNames) {
private val name: String

private var status: String? = null


init {
this.name = name.name
}


fun WriteToFile(data: String?) {
var data = data
try {
if (data == null)
data = ""

val fileOutput = context.openFileOutput(name, Context.MODE_PRIVATE)

val writer = OutputStreamWriter(fileOutput)

writer.write(data)
writer.close()
} catch (e: IOException) {
status = context.getString(R.string.error_file_write) + ": " + e.toString()
}

}


fun ReadFromFile(): String {
var result = ""

try {
val inputStream = context.openFileInput(name)

if (inputStream != null) {
val inputStreamReader = InputStreamReader(inputStream)
val bufferedReader = BufferedReader(inputStreamReader)

var line: String
val allContent = StringBuilder()

while ((line = bufferedReader.readLine()) != null) {
    allContent.append(line)
}

inputStream.close()

result = allContent.toString()
}
} catch (e: FileNotFoundException) {
status = context.getString(R.string.error_file_not_found) + ": " + e.toString()
} catch (e: IOException) {
status = context.getString(R.string.error_file_read) + ": " + e.toString()
}

return result
}

companion object {
internal val extension = ".dfm"
}


}
*/